### DISCUSSION TITLE
compile error

### DISCUSSION THREAD

**Discussion by blazerye** [rdkit/rdkit#4261] [Q&A] [answered]
**Describe the bug**
building from source, but compile failed
![image](https://user-images.githubusercontent.com/16797849/122359575-3fee0f00-cf88-11eb-9b3b-4a7165204089.png)

**To Reproduce**
building commands
`
RDBASE=../rdkit

export LD_LIBRARY_PATH=$RDBASE/lib

cd $RDBASE

mkdir build

cd build

cmake -D PYTHON_LIBRARY=/usr/lib/python3.5/config-3.5m-x86_64-linux-gnu/libpython3.5.so \
      -D PYTHON_INCLUDE_DIR=/usr/include/python3.5/ -D PYTHON_EXECUTABLE=/usr/bin/python3 ..

make -j32

make install -j32
`

**Expected behavior**
no compile error

**Screenshots**
![image](https://user-images.githubusercontent.com/16797849/122349822-30b69380-cf7f-11eb-9ded-82a4f5d2ed16.png)

**Configuration (please complete the following information):**
 - RDKit version: 2021.03.3 (download code from master branch yesterday)
 - OS: Ubuntu 16.04
 - Python version (if relevant): 3.5
 - Are you using conda? no
 - If you are using conda, which channel did you install the rdkit from?
 - If you are not using conda: how did you install the RDKit? build from source

**Additional context**
no

**Comment by ptosco** [comment]
@blazerygy You do not report the compiler version you are using so I cannot say this for sure, but it looks like you might be using a `gcc` version older than 5; see these similar reports:

https://github.com/google/shaka-packager/issues/694
https://github.com/google/draco/issues/582

If that's indeed the case, your problem can be solved by upgrading the compiler version.

**Comment by greglandrum** [comment]
@blazerygy : are you using v2021.03.3 or the master branch?

**Comment by blazerye** [comment]
> @blazerygy You do not report the compiler version you are using so I cannot say this for sure, but it looks like you might be using a `gcc` version older than 5; see these similar reports:
> 
> [google/shaka-packager#694](https://github.com/google/shaka-packager/issues/694)
> [google/draco#582](https://github.com/google/draco/issues/582)
> 
> If that's indeed the case, your problem can be solved by upgrading the compiler version.

gcc：gcc version 4.9.3 (Ubuntu 4.9.3-13ubuntu2)

and then, I update gcc to 7.4.0, still compile error
![image](https://user-images.githubusercontent.com/16797849/122384580-6ae35d80-cf9e-11eb-9082-e98c047abb18.png)
![image](https://user-images.githubusercontent.com/16797849/122384411-412a3680-cf9e-11eb-8fb8-013721d13c1e.png)

**Comment by blazerye** [comment]
> @blazerygy : are you using v2021.03.3 or the master branch?

the master branch

**Comment by ptosco** [comment, marked as answer]
@blazerygy can you check that `cmake` actually picked up that compiler:
```
$ grep 'CMAKE_.*_COMPILER:FILEPATH' CMakeCache.txt
```

**Comment by blazerye** [comment]
> @blazerygy can you check that `cmake` actually picked up that compiler:
> 
> ```
> $ grep 'CMAKE_.*_COMPILER:FILEPATH' CMakeCache.txt
> ```

The gcc used by cmake is indeed a low version. After I updated it to version 7.4, the problem was solved, thx @ptosco
